clang/lib/Driver/ToolChains/Darwin.cpp

```diff
@@ -1645,14 +1645,15 @@ void DarwinClang::AddLinkRuntimeLibArgs(const ArgList &Args,
     CmdArgs.push_back("-lSystem");
 
   // Select the dynamic runtime library and the target specific static library.
-  if (isTargetIOSBased()) {
-    // If we are compiling as iOS / simulator, don't attempt to link libgcc_s.1,
-    // it never went into the SDK.
-    // Linking against libgcc_s.1 isn't needed for iOS 5.0+
-    if (isIPhoneOSVersionLT(5, 0) && !isTargetIOSSimulator() &&
-        getTriple().getArch() != llvm::Triple::aarch64)
-      CmdArgs.push_back("-lgcc_s.1");
-  }
+  // If we are compiling as iOS / simulator, don't attempt to link libgcc_s.1,
+  // it never went into the SDK.
+  // Linking against libgcc_s.1 isn't needed for iOS 5.0+ or macOS 10.6+
+  if (isTargetIOSBased() && isIPhoneOSVersionLT(5, 0) &&
+      !isTargetIOSSimulator() && getTriple().getArch() != llvm::Triple::aarch64)
+    CmdArgs.push_back("-lgcc_s.1");
+  else if (isTargetMacOSBased() && isMacosxVersionLT(10, 6) &&
+           getTriple().getArch() != llvm::Triple::aarch64)
```

Comment:
```suggestion
  if (isTargetMacOSBased() && isMacosxVersionLT(10, 6) &&
      getTriple().getArch() != llvm::Triple::aarch64)
```
Doesn't seem like you really need `else` here? Seems more readable to drop it. You could probably also stick both `if`s under `if (getTriple().getArch() != llvm::Triple::aarch64)`.

Reply:
If target is iOS based it's impossible for it to be macOS based, so the else should speed things up on iOS by not running isTargetMacOSBased wouldn't it?

Reply:
Could also do one big if statement.
```c++
if (getTriple().getArch() != llvm::Triple::aarch64 &&
    ((isTargetIOSBased() && isIPhoneOSVersionLT(5, 0) &&
      !isTargetIOSSimulator()) ||
     (isTargetMacOSBased() && isMacosxVersionLT(10, 6))))
  CmdArgs.push_back("-lgcc_s.1");
```